### PR TITLE
Override password manager icons

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/33242

Note: This is a first pass of the icon. @aguscruiz wants to land this now, but the icon might be changed in the not too distant future.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
@bridiver ptal

**Review by mherrmann — APPROVED** [review]
@fallaciousreasoning I'm not sure why you requested my review because this is not code I'm very familiar with; But lgtm.

**Comment by fallaciousreasoning** [comment]
Sorry @mherrmann - you were listed as a suggested reviewer by Github :/ @bridiver could you take a look?

File: build/commands/lib/util.js [hunk 1]
```diff
@@ -284,6 +284,12 @@ const util = {
       path.join(config.srcDir, 'ui', 'webui', 'resources', 'images',
           'chrome_logo_dark.svg')])
 
+    // Override Password Manager icons
```
Comment (bridiver) [inline build/commands/lib/util.js:287]: This is really not a great way to handle this stuff and we need to stop using this. In this particular case you can do a chromium_src override for password_manager_ui.cc 
```
#define SetupWebUIDataSource(...) SetupWebUIDataSource(__VA_ARGS__); source->AddResourcePath("images/password_manager_logo.svg",
                          IDR_BRAVE_PASSWORD_MANAGER_LOGO);
```
where `IDR_BRAVE_PASSWORD_MANAGER_LOGO` goes in `brave_unscaled_resources.grd`.

Comment (bridiver) [reply build/commands/lib/util.js:287]: fyi - this does the same thing as the google chrome branding in this method
```
#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
  // Overwrite ubranded logo for Chrome-branded builds.
  source->AddResourcePath("images/password_manager_logo.svg",
                          IDR_CHROME_PASSWORD_MANAGER_LOGO);

  // This path is used in the manifest of the PasswordManager web app
  // (chrome/browser/resources/password_manager/
  // chrome_branded_manifest.webmanifest).
  source->AddResourcePath("images/password_manager_pwa_icon.svg",
                          IDR_CHROME_PASSWORD_MANAGER_PWA_ICON);
#endif
```

Comment (fallaciousreasoning) [reply build/commands/lib/util.js]: Cool, thanks @bridiver - I was just cargo culting the other stuff :)

**Review by bridiver — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`grd`++

**Pull request merged**